content/_layouts/docs.njk

```diff
@@ -0,0 +1,92 @@
+---
+layout: base
+---
+
+{# Macros ------------------------------- #}
+
+{% macro describe(pattern) %}
+  {% if pattern.note %}
+    {{ pattern.note | md | safe }}
+  {% endif %}
+
+  {% if pattern.params %}
+    <h4>Params:</h4>
+    <ul data-dox="params">
+      {% for key, item in pattern.params %}
+        <li>
+          <code>
+            <strong>{{ key }}</strong>
+            {% if item.type %}
+              {{ '{' + item.type + '}' }}
+            {% endif %}
+            {% if item.default %}
+              = <em>{{ item.default }}</em>
+            {% endif %}
+          </code>
+          {% if item.note %}
+            {{ item.note | md | safe }}
+          {% endif %}
+        </li>
+      {% endfor %}
+    </ul>
+  {% endif %}
+
+  {% if pattern.links %}
+    <h4>Links:</h4>
+    <ul>
+      {% for link in pattern.links %}
+        <li>{{ link | mdInline | safe }}</li>
+      {% endfor %}
+    </ul>
+  {% endif %}
+
+  {% if pattern.example %}
+    <h4>Example:</h4>
+    {% highlight doc_lang %}{{ pattern.example | safe }}{% endhighlight %}
+  {% endif %}
+
+  {% if pattern.todo %}
+    <h4>Todo:</h4>
+    {{ pattern.todo | md | safe }}
+  {% endif %}
+{% endmacro %}
+
+{% macro show(
+  patterns
+) %}
+  {% for pattern in patterns %}
+    <article id="{{ pattern.slug }}">
```

Comment:
Should we bump all these h3s and h4s up since there are no H2s on these doc pages?

Reply:
OH, I take that back. some pages don't have an H2 until "see also" because the content on the page doesn't have a "category" assigned like the `.../sample/macros/contact/` page, but `...sample/macros/embed/` has categories that become H2s like "Patterns", "Embed" "Layout". Should we always add a category even if it is a shared one by all items on that page? Or should we look for a category and if there isn't one, make the heading level one up from the pages with categories? That option seems more complicated.

Reply:
I'm open to either of these options. There's already special handling for no categories, so that one isn't actually more complicated. Don't have an idea for what  it should say, though. Maybe just "Macros"?

Reply:
We lost the header I think now: 
![missing-oddbird-header-local-vs-preview](https://user-images.githubusercontent.com/1581694/72827379-4d428500-3c40-11ea-96e3-c757fa5b35bf.jpg)

Reply:
I think this should be fixed, if one of you wants to confirm.